app/api.py

```diff
@@ -297,6 +301,37 @@ def get(self):
             self, ApiActionLog.READ, len(records), len(notes))
 
 
+class PhotoUpload(utils.BaseHandler):
+    https_required = True
+
+    def post(self):
+        if not (self.auth):
+            self.info(
+                403,
+                message='Missing or invalid authorization key',
+                style='plain')
+            return
+        photo = self.request.body_file
+        if photo is not None:
+            photo = photo.read()
+            photo = images.Image(photo)
+            photo, photo_url = create_photo(photo, self)
+            if photo:
+                photo.put()
+        else :
+            self.info(
+                403,
+                message='Photo URL is wrong',
+                style='plain')
+            return
+
+        self.response.headers['Content-Type'] = 'application/xml'
+        self.write('<?xml version="1.0"?>\n')
+        self.write('<response><url>')
+        self.write(photo_url)
```

Comment:
Can you apply XML escaping just in case? It may contain "&" which should be escaped.